src/util.ts

```diff
@@ -92,3 +92,16 @@ export function clearBusyState(...elements: Element[]) {
     element.removeAttribute("aria-busy")
   }
 }
+
+export function waitForLoad(element: HTMLLinkElement): Promise<void> {
+  return new Promise((resolve) => {
+    const onComplete = () => {
+      element.removeEventListener("error", onComplete)
+      element.removeEventListener("load", onComplete)
+      resolve()
+    }
+
```

Comment:
Could `setTimeout(resolve, resourceLoadingTimeoutInMilliseconds)` serve as a guard against remote resources that take forever to load?